sql/insert.go

```diff
@@ -186,3 +224,94 @@ func (p *planner) processColumns(tableDesc *TableDescriptor,
 
 	return cols, nil
 }
+
+func (p *planner) makeDefaultExprs(cols []ColumnDescriptor) ([]parser.Expr, error) {
+	// Check to see if any of the columns have DEFAULT expressions. If there are
+	// no DEFAULT expressions, we don't bother with constructing the defaults map
+	// as the defaults are all NULL.
+	haveDefaults := false
+	for _, col := range cols {
+		if col.DefaultExpr != nil {
+			haveDefaults = true
+			break
+		}
+	}
+	if !haveDefaults {
+		return nil, nil
+	}
+
+	// Construct a SELECT statement with an output for each column. The output is
+	// either the DEFAULT expression or NULL if no default expression was
+	// configured.
+	var buf bytes.Buffer
+	buf.WriteString("SELECT ")
+	for i, col := range cols {
+		if i > 0 {
+			buf.WriteString(", ")
+		}
+		if col.DefaultExpr != nil {
+			buf.WriteString(*col.DefaultExpr)
+		} else {
+			buf.WriteString("NULL")
+		}
+	}
+
+	sql := buf.String()
+	stmts, err := parser.Parse(sql, parser.Syntax(p.session.Syntax))
```

Comment:
This makes my sql-injection sense twitch. Can we not build up the AST directly instead of building a string and parsing it? 

`p.session.Syntax` is incorrect: the default expressions must be parsed in the language context in which they were defined, not the context of the session that causes them to be evaluated. (Unless we can guarantee that defaults are limited to expressions that are parsed identically in all syntaxes, in which case it would still be better to just hard-code one syntax here instead of using the session)